packages/grid/_modules_/grid/components/cell/GridEditInputCell.tsx

```diff
@@ -22,36 +22,44 @@ export function GridEditInputCell(props: GridCellParams & InputBaseProps) {
   } = props;
 
   const [valueState, setValueState] = React.useState(value);
+  const inputType = mapColDefTypeToInputType(colDef.type);
 
   const handleChange = React.useCallback(
     (event) => {
       const newValue = event.target.value;
       const editProps = {
-        value: colDef.type === 'date' || colDef.type === 'dateTime' ? new Date(newValue) : newValue,
+        value: newValue,
       };
+
+      if (colDef.type === 'date' || colDef.type === 'dateTime') {
+        if (newValue === '') {
```

Comment:
This should be a ternary operator to avoid this nested if.

Reply:
Sounds good, I did this in a rush, didn't try yo make it polished :)